mexico-city/src/mc_nitro.rs

```diff
@@ -0,0 +1,219 @@
+            },
+            MCMessage::CloseTLSSession(session_id) => {
+                println!("mc_nitro::main CloseTLSSession");
+                let cs_result = managers::baja_manager::close_session(session_id);
+                let return_message: MCMessage = match cs_result {
+                    Ok(_) => MCMessage::Status(NitroStatus::Success),
+                    Err(_) => MCMessage::Status(NitroStatus::Fail),
+                };
+                return_message
+            },
+            MCMessage::GetTLSDataNeeded(session_id) => {
+                println!("mc_nitro::main GetTLSDataNeeded");
+                let return_message = match managers::baja_manager::get_data_needed(session_id) {
+                    Ok(needed) => MCMessage::TLSDataNeeded(needed),
+                    Err(_) => MCMessage::Status(NitroStatus::Fail),
+                };
+                return_message
+            },
+            MCMessage::SendTLSData(session_id, tls_data) => {
+                println!("mc_nitro::main SendTLSData");
+                let return_message = match managers::baja_manager::send_data(session_id, &tls_data) {
+                    Ok(_) => MCMessage::Status(NitroStatus::Success),
+                    Err(_) => MCMessage::Status(NitroStatus::Fail),
+                };
+                return_message
+            },
+            MCMessage::GetTLSData(session_id) => {
+                println!("mc_nitro::main GetTLSData");
+                let return_message = match managers::baja_manager::get_data(session_id) {
+                    Ok((active, output_data)) => MCMessage::TLSData(output_data, active),
+                    Err(_) => MCMessage::Status(NitroStatus::Fail),
+                };
+                return_message
+            },
+            MCMessage::GetPSAAttestationToken(challenge) => {
+                println!("mc_nitro::main GetPSAAttestationToken");
+                get_psa_attestation_token(&challenge)
+                    .map_err(|err| format!("mc_nitro::nitro_main get_psa_attestation_failed:{:?}", err))?
+            },
+            MCMessage::ResetEnclave => {
+                println!("mc_nitro::main ResetEnclave");
+                MCMessage::Status(NitroStatus::Success)
+            },
+            _ => {
+                println!("mc_nitro::main Unknown Opcode");
+                MCMessage::Status(NitroStatus::Unimplemented)
+            },
+        };
+        let return_buffer = bincode::serialize(&return_message)
+            .map_err(|err| format!("mc_nitro::main serialize failed:{:?}", err))?;
+        println!("mc_nitro::main calling send buffer with buffer_len:{:?}", return_buffer.len());
+        send_buffer(fd, &return_buffer)
+            .map_err(|err| format!("mc_nitro::main send_buffer failed:{:?}", err))?;
+    }
+}
+
+fn initialize(policy_json: &String) -> Result<MCMessage, String> {
+    println!("mc_nitro::initialize started");
+    managers::baja_manager::init_baja(&policy_json)
```

Comment:
What is the type of `init_baja`?  The parameter `policy_json` is already a reference to a string, and a second reference is being taken here.  Further, is `&String` the correct type for `policy_json`, or will `&str` do?

Reply:
The input parameter type of `init_baja` is `&str`. Taking `&&String` does appear to be valid in this case (`&String` gives `str`, `&&String` gives `&str`).

However, your point is taken. It would probably be clearer for `fn initialize` to take a `&str`. I look into doing this and get back to you.